lib/_http_agent.js

```diff
@@ -217,7 +217,7 @@ Agent.defaultMaxSockets = Infinity;
 Agent.prototype.createConnection = net.createConnection;
 
 // Get the key for a given set of request options
-Agent.prototype.getName = function getName(options) {
+Agent.prototype.getName = function getName(options = {}) {
   let name = options.host || 'localhost';
```

Comment:
I'd prefer if instead of creating a throw away object, we used optional chaining and set/keep the default value to `undefined`.
```suggestion
Agent.prototype.getName = function getName(options = undefined) {
  let name = options?.host || 'localhost';
```

Reply:
The `option` has many attributes, such as `port`,`localAddress`, `family` and so on. It will be more simplify to create a default object than make every attributes with `?.`

Reply:
> It will be more simplify to create a default object than make every attributes with `?.`

Depends what you mean by simplify, it's a smaller change in term of source code changes, but it's not simpler for end users. Using a throw away objects also has side effects which could be surprising for them:

```js
// In userland:
Object.prototype.host = 'something unrelated';

// In core:
function getName(options = {}) { return options.host || 'localhost'; }
console.log(getName()); // 'something unrelated'
```

```js
// In userland:
Object.prototype.host = 'something unrelated';

// In core:
function getName(options = undefined) { return options?.host || 'localhost'; }
console.log(getName()); // 'localhost'
```

My opinion is that Node.js core modules should use optional chaining over creating a throw away objects, it's not a blocking concern by any mean though. Feel free to disagree, the PR is fine as is.